### PR TITLE
guide page on classes; correction and expansion

### PR DESCRIPTION
**Pull request by paddywwoof**
In the guide there seems to be no explanation or example of using PyModule.add_class().

The explanation and example of using `#[args(...)]` is very thin (and missing some `"`s) the use of arg and args as parameters(?) method arguments(?) is quite confusing. Also the example method doesn't do anything so it's almost impossible to discern how the different (very powerful) functionality can be used.

### CODE REVIEW

File: guide/src/class.md [hunk 1]
```diff
@@ -13,7 +13,14 @@ struct MyClass {
    num: i32,
    debug: bool,
 }
+
+#[pymodule]
```
Comment (kngwyu) [inline guide/src/class.md]: I think this should be in another subsection, e.g., `## Add class to module`

File: guide/src/class.md [hunk 2]
```diff
@@ -460,13 +478,54 @@ use pyo3::types::{PyDict, PyTuple};
...
+    fn new(obj: &PyRawObject, num: i32, debug: bool) {
+        obj.init({
+            MyClass {
+                num,
+                debug,
+            }
+        });
+    }
+    #[args(num="10", debug="true", py_args="*", name="\"Hello\"", py_kwargs="**")]
+    fn method(&mut self, num: i32, debug: bool, name: &str, py_args: &PyTuple,
```
Comment (kngwyu) [inline guide/src/class.md]: Could you please format this method in rustfmt style?
```rust
    #[args(
        num = "10",
        debug = "true",
        py_args = "*",
        name = "\"Hello\"",
        py_kwargs = "**"
    )]
    fn method(
        &mut self,
        num: i32,
        debug: bool,
        name: &str,
        py_args: &PyTuple,
        py_kwargs: Option<&PyDict>,
    ) -> PyResult<String> {
        ...
    }
```

File: guide/src/class.md [hunk 3]
```diff
@@ -460,13 +478,54 @@ use pyo3::types::{PyDict, PyTuple};
...
 }
-```
 
+#[pymodule]
+fn mymodule(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<MyClass>()?;
+    Ok(())
+}
+```
+N.B. the position of the `"*"` argument (if included) controls the system of handling positional and keyword arguments. In Python:
```
Comment (kngwyu) [inline guide/src/class.md]: This Python code example is really good :+1:

File: guide/src/class.md [hunk 4]
```diff
@@ -460,13 +478,54 @@ use pyo3::types::{PyDict, PyTuple};
...
+    fn make_change(&mut self, num: i32, debug: bool) -> PyResult<String> {
+        self.num = num;
+        self.debug = debug;
+        Ok(format!("num={}, debug={}", self.num, self.debug))
     }
 }
-```
 
+#[pymodule]
+fn mymodule(_py: Python, m: &PyModule) -> PyResult<()> {
```
Comment (kngwyu) [inline guide/src/class.md]: I think this is unrelated here

**Comment by kngwyu** [comment]
Thank you, they are missing parts for us.

File: guide/src/class.md (hunk 3 above)
Comment (paddywwoof) [reply guide/src/class.md]: Hi, all good suggestions. Hopefully I've incorporated them all.

**Comment by kngwyu** [comment]
Thank you!
